blocks/bricks/attention.py

```diff
@@ -605,8 +605,11 @@ def take_glimpses(self, **kwargs):
             kwargs.pop(self.preprocessed_attended_name, None),
             kwargs.pop(self.attended_mask_name, None),
             **dict_union(states, glimpses_needed))
-        if kwargs:
-            raise ValueError("extra args to take_glimpses: {}".format(kwargs))
+        # The following check is too excessive.
+        # there could be context information which is needed by the transition
+        # apply, but is also passed to this method.
+        # if kwargs:
+        #    raise ValueError("extra args to take_glimpses: {}".format(kwargs))
```

Comment:
I see. May I ask you to remove lines 611-612 and change the comment. It's not a good practice to leave commented out code.

And can you add a simple test which crashes if this check is not commented out. I suppose it may be a simple recurrent transition with some dummy context.